src/components/clusterConfiguration/ClusterSshKeyField.tsx

```diff
@@ -0,0 +1,67 @@
+import * as React from 'react';
+import { useField } from 'formik';
+import { FormGroup, Switch } from '@patternfly/react-core';
+import { FieldProps } from '../ui/formik/types';
+import { getFieldId } from '../ui/formik/utils';
+import HelperText from '../ui/formik/HelperText';
+import TextAreaField from '../ui/formik/TextAreaField';
+
+import './ClusterSshKeyField.css';
+
+interface ClusterSshKeyFieldProps extends FieldProps {
+  isSwitchHidden: boolean;
+  onToggle: (isChecked: boolean) => void;
+  onClusterSshKeyVisibilityChanged: () => void;
+}
+
+const sshPublicKeyHelperText = (
+  <>
+    Provide a SSH public key to debug OpenShift nodes after installation. To generate a new key, use
+    the <em>ssh-keygen -o</em> command and paste the value of <em>~/.ssh/id_rsa.pub</em> here.
+  </>
+);
+
+const label = 'Host SSH Public Key for troubleshooting after installation';
+
+const ClusterSshKeyField: React.FC<ClusterSshKeyFieldProps> = ({
```

Comment:
This component handles so specific use-case that there is probably no reason to make it generic